pkg/shell/base_index.js

```diff
@@ -496,7 +498,84 @@ function Index() {
     if (typeof self.navigate !== "function")
         throw Error("Index requires a prototype with a navigate function");
 
-    self.frames = new Frames(self);
+    /* Session timing out after inactivity */
+    let session_timer = null;
+    let session_final_timer = null;
+    let session_timeout = 0;
+    let final_countdown = 30000; // last 30 seconds
```

Comment:
di-di-diiiiii-diiiii! ♩ ♪ ♫

Reply:
I wonder if it should be a minute, not 30 seconds.

People may leave their browser in the background, have a ton of tabs, and so on. It might take them some time to get back to Cockpit to see the window, even if they're trying to do so.

(I am pretty sure our user base would be much more likely to have lots of windows and browser tabs open than the general population. Considering that Firefox scrolls tabs off the edge (but has an overflow dropdown and Chrome just shrinks the tabs until there's no icon or text or anything useful, it could take some time.)

But both a minute and 30 seconds are kind of arbitrary. Any timeframe would be. However, 30 seconds may feel a little rushed under some circumstances.

Reply:
Right, it's arbitrary, but then again this is also not really important -- if you don't pay attention, you can just log back in.

Reply:
Yeah, that's true.

I wanted to make sure someone has enough time to sign back in in case they may have mobility issues or are multi-tasking.

This becomes an issue especially if they're in the middle of an action that isn't complete or a background task. (Thankfully, Cockpit doesn't have too many actions like this.)

Reply:
We discussed that a while ago -- any action that takes a longer time must not depend on the cockpit UI running all the time. E. g. formatting a storage device or installing updates all start a background task, and you can log out of cockpit and back in to continue.